src/main/java/school/lemon/changerequest/java/introduction/hw2/Task9.java

```diff
@@ -6,37 +6,51 @@
 
 public class Task9 {
     public static String[] extractData(String URL) {
-        String[] arr = {"", "", ""};
-
-        Pattern patternSchema = Pattern.compile("([a-z]*://)");
-        Matcher matcherSchema = patternSchema.matcher(URL);
-        if (matcherSchema.find()) {
-            Pattern p = Pattern.compile("[a-z]*");
-            Matcher m = p.matcher(matcherSchema.group());
-            if (m.find()) {
-                arr[0] = m.group();
-            }
-        }
 
-        Pattern patternHost = Pattern.compile("([\\w]+\\.){1}([\\w]+\\.?)+");
-        Matcher matcherHost1 = patternHost.matcher(URL);
-        if (matcherHost1.find()) {
-            arr[1] = matcherHost1.group();
-        }
+        if (URL == null) return new String[]{"", "", ""};
+        String regexp = "(?:^([a-z]+)://)?([\\da-z._-]+)(?::(\\d+)?)?.*";
+        Pattern pattern = Pattern.compile(regexp);
+        Matcher matcher = pattern.matcher(URL);
 
-        Pattern patternPort = Pattern.compile("(:[0-9]+)");
-        Matcher matcherPort = patternPort.matcher(URL);
-        if (matcherPort.find()) {
-            Pattern pattern11 = Pattern.compile("([0-9]+)");
-            Matcher matcher11 = pattern11.matcher(matcherPort.group());
-            if (matcher11.find()) {
-                arr[2] = matcher11.group();
-            }
+        if (matcher.matches()) {
+            String schema = matcher.group(1);
+            String host = matcher.group(2);
+            String port = matcher.group(3);
+            return new String[]{schema != null ? schema : "", host != null ? host : "", port != null ? port : ""};
         }
-
-        return arr;
+        return new String[]{"", "", ""};
     }
 
+
+//        Pattern patternSchema = Pattern.compile("([a-z]*://)");
+//        Matcher matcherSchema = patternSchema.matcher(URL);
+//        if (matcherSchema.find()) {
+//            Pattern p = Pattern.compile("[a-z]*");
+//            Matcher m = p.matcher(matcherSchema.group());
+//            if (m.find()) {
+//                arr[0] = m.group();
+//            }
+//        }
+//
+//        Pattern patternHost = Pattern.compile("([\\w]+\\.){1}([\\w]+\\.?)+");
+//        Matcher matcherHost1 = patternHost.matcher(URL);
+//        if (matcherHost1.find()) {
+//            arr[1] = matcherHost1.group();
+//        }
+//
+//        Pattern patternPort = Pattern.compile("(:[0-9]+)");
+//        Matcher matcherPort = patternPort.matcher(URL);
+//        if (matcherPort.find()) {
+//            Pattern pattern11 = Pattern.compile("([0-9]+)");
+//            Matcher matcher11 = pattern11.matcher(matcherPort.group());
+//            if (matcher11.find()) {
+//                arr[2] = matcher11.group();
+//            }
+//        }
+//
+//        return arr;
```

Comment:
do not commit comments, there is no need for them, cause version control system will anyway track all your changes.